tree.go

```diff
@@ -208,14 +208,13 @@ func NewLeafNode(stem []byte, values [][]byte) *LeafNode {
 		c1poly[EmptyCodeHashFirstHalfIdx].Equal(&EmptyCodeHashFirstHalfValue) &&
 		c1poly[EmptyCodeHashSecondHalfIdx].Equal(&EmptyCodeHashSecondHalfValue)
 	if containsEmptyCodeHash {
-		// We start c1 with the empty code hash point.
-		c1 = EmptyCodeHashPoint
```

Comment:
Before `EmptyCodeHashPoint` was a `*Point`, so this was doing a pointer copy and then in L218 (deleted line) it was mutating the value of that global var. 

Pretty shocking this didn't become apparent in tests in this repo nor a different hash in the replay bench. :/

Reply:
indeed